runhouse/resources/hardware/cluster.py

```diff
@@ -1127,6 +1128,33 @@ def _start_or_restart_helper(
         # Update the cluster config on the cluster
         self.save_config_to_cluster()
 
+        # Save a limited version of the local ~/.rh config to the cluster with the user's hashed token,
+        # if such does not exist on the cluster
+        if rns_client.token:
+            user_config = yaml.safe_dump(
+                {
+                    "token": rns_client.cluster_token(
+                        resource_address=rns_client.username
+                    ),
+                    "username": rns_client.username,
+                    "default_folder": rns_client.default_folder,
+                }
+            )
+
+            if (
+                self.run(
+                    ["[ -f ~/.rh/config.yaml ]"],
+                    node=self.head_ip,
+                    require_outputs=False,
+                )[0]
+                == 0
+            ):
+                logger.debug("Did not change config.yaml")
+            else:
+                command = f"echo '{user_config}' > ~/.rh/config.yaml"
```

Comment:
we need to change this to a proper rsync rather than the `echo xxx`, which will log and print the token